### PR TITLE
simple-http-server: init at 0.6.1

### PR DESCRIPTION
**Pull request by Mephistophiles**
Signed-off-by: Maxim Zhukov <mussitantesmortem@gmail.com>

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 120587` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>simple-http-server</li>
  </ul>
</details>

File: pkgs/servers/simple-http-server/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  src = fetchFromGitHub {
+    owner = "TheWaWaR";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "01a129i1ph3m8k6zkdcqnnkqbhlqpk7qvvdsz2i2kas54csbgsww";
+  };
+
+  cargoSha256 = "050avk6wff8v1dlsfvxwvldmmgfakdxmhglv2bhvc2f3q8cf1d5d";
+
+  nativeBuildInputs = [ pkg-config makeWrapper ];
```
Comment (legendofmiracles) [inline pkgs/servers/simple-http-server/default.nix:16]: You don't need makeWrapper as a build dependency

**Review by legendofmiracles — CHANGES_REQUESTED** [review]

File: pkgs/servers/simple-http-server/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  buildInputs = [ openssl ];
+
+  # Currently no tests are implemented, so we avoid building the package twice
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Simple http server in Rust (Windows/Mac/Linux)";
+    homepage = "https://github.com/TheWaWaR/simple-http-server";
+    license = licenses.mit;
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (legendofmiracles) [inline pkgs/servers/simple-http-server/default.nix:27]: Doesn't simple-http-server support all general platforms? This should be set by buildRustPackage.

**Comment by Ma27** [comment]
@Mephistophiles why am I listed as maintainer here?

File: pkgs/servers/simple-http-server/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    maintainers = with maintainers; [ backuitist globin ma27 ];
```
Comment (fabaff) [inline pkgs/servers/simple-http-server/default.nix:28]: Do does people agree to maintain the package? At first glance, I would say that this is copy-&-paste error.

File: pkgs/servers/simple-http-server/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  nativeBuildInputs = [ pkg-config makeWrapper ];
+
+  buildInputs = [ openssl ];
+
+  # Currently no tests are implemented, so we avoid building the package twice
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Simple http server in Rust (Windows/Mac/Linux)";
```
Comment (fabaff) [inline pkgs/servers/simple-http-server/default.nix:24]: Opinionated change :wink: 

```suggestion
    description = "Simple HTTP server in Rust";
```

File: pkgs/servers/simple-http-server/default.nix (hunk 3 above)
Comment (legendofmiracles) [reply pkgs/servers/simple-http-server/default.nix:28]: Just like the makeWrapper part

File: pkgs/servers/simple-http-server/default.nix (hunk 1 above)
Comment (Mephistophiles) [reply pkgs/servers/simple-http-server/default.nix]: Fixed, Thanks!

File: pkgs/servers/simple-http-server/default.nix (hunk 2 above)
Comment (Mephistophiles) [reply pkgs/servers/simple-http-server/default.nix]: Fixed. Thanks

**Comment by Mephistophiles** [comment]
> @Mephistophiles why am I listed as maintainer here?

ooops, sorry! copy-paste error >_<.  Fixed

**Comment by Mephistophiles** [comment]
Thanks everyone for your review!

**Review by legendofmiracles — APPROVED** [review]
LGTM now

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 120587` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>simple-http-server</li>
  </ul>
</details>

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 120587` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>simple-http-server</li>
  </ul>
</details>

**Review by davidak — APPROVED** [review]
Change looks good

Builds, works

**Pull request merged**